dll/win32/shell32/wine/shellpath.c

```diff
@@ -87,8 +87,8 @@ DoGetProductType(PNT_PRODUCT_TYPE ProductType)
         return FALSE;
 
     cbValue = sizeof(szValue);
-    error = RegQueryValueExW(hKey, L"ProductType", NULL, NULL, (LPBYTE)szValue, &cbValue);
-    if (!error)
+    error = RegQueryValueExW(hKey, L"ProductType", NULL, &dwType, (LPBYTE)szValue, &cbValue);
+    if (!error && dwType == REG_SZ)
     {
         if (lstrcmpW(szValue, L"WinNT") == 0)
             *ProductType = NtProductWinNt;
```

Comment:
@katahiromz (for later): missing check for type NtProductServer.

Reply:
`NtProductServer` is already set as default value...

Reply:
Oh yes that's true.